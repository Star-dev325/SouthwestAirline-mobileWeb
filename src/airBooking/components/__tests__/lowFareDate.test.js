import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import FakeClock from 'test/unit/helpers/fakeClock';
import { render, fireEvent } from '@testing-library/react';
import LowFareDate from 'src/airBooking/components/lowFareDate';

describe('LowFareDate', () => {
  let onClickCalendarIconFnStub;

  beforeEach(() => {
    const noop = () => {};

    onClickCalendarIconFnStub = jest.fn(noop);
  });

  afterEach(() => {
    FakeClock.restore();
    jest.clearAllMocks();
  });

  it('should mount the low fare date props', () => {
    const { baseElement } = createComponent();

    expect(baseElement).toMatchSnapshot();
  });

  it('should show a pretty dayjs styled date', () => {
    const { container } = createComponent();

    expect(container).toHaveTextContent('Depart Sun, Apr 5, 2020');
  });

  it('should show returning instead of departs', () => {
    const { container } = createComponent({ isInbound: true });

    expect(container).toHaveTextContent('Return Sun, Apr 5, 2020');
  });

  it('should show returning icon', () => {
    const { baseElement } = createComponent({ isInbound: true });

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onClickCalendarIconFnStub when clicking calendar icon', () => {
    const { container } = createComponent();

    fireEvent.click(container.querySelector('.icon_calender'));
    expect(onClickCalendarIconFnStub).toHaveBeenCalled();
  });

  const createComponent = (props = {}) => {
    const defaultProps = {
      flightDate: '2020-04-05',
      isInbound: false,
      onClickCalendarIconFn: onClickCalendarIconFnStub
    };

    return render(<LowFareDate {...defaultProps} {...props} />);
  };
});
