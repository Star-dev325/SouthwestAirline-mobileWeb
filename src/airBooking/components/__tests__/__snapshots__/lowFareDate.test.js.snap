// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LowFareDate should mount the low fare date props 1`] = `
<body>
  <div>
    <div
      class="low-fare-date"
    >
      <div
        class="selected-date"
      >
        <div
          class="flight-logo-wrapper svg-wrapper low-fare--col-icon"
        >
          <i
            class="full-width-svg icon icon_airplane-depart"
          />
        </div>
        <div
          class="flight-label-wrapper"
        >
          <div
            class="label-container"
          >
            <div
              class="label-container--label"
            >
              Depart 
            </div>
            <div
              class="label-container--content"
            >
              Sun, Apr 5, 2020
            </div>
          </div>
        </div>
      </div>
      <div
        class="calendar-select"
      >
        <i
          class="full-width-svg icon icon_calender"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`LowFareDate should show returning icon 1`] = `
<body>
  <div>
    <div
      class="low-fare-date"
    >
      <div
        class="selected-date"
      >
        <div
          class="flight-logo-wrapper svg-wrapper low-fare--col-icon-return"
        >
          <i
            class="full-width-svg icon icon_airplane-return"
          />
        </div>
        <div
          class="flight-label-wrapper"
        >
          <div
            class="label-container"
          >
            <div
              class="label-container--label"
            >
              Return 
            </div>
            <div
              class="label-container--content"
            >
              Sun, Apr 5, 2020
            </div>
          </div>
        </div>
      </div>
      <div
        class="calendar-select"
      >
        <i
          class="full-width-svg icon icon_calender"
        />
      </div>
    </div>
  </div>
</body>
`;
